Employee attrition problem\
Employees are very important in the organization. Organization's performance is highly based on the quality of the employees\
An Organization can face various challenges due employee attrition, and they are:

1. Expensive in terms of both money and time to train new employees.
2. Loss of experienced employees
3. Impact in productivity
4. Impact profit

Business Questions:
1. What factors are contributing more to employee attrition?
2. What type of measures should the company take in order to retain their employees?

Age
Department
DailyRate
Gender
Education
YearsSinceLastPromotion
WorkLifeBalanceRelationshipSatisfaction
PerformanceRating
MaritalStatus
JobSatisfaction
JobInvolvement
EnvironmentSatisfaction
EducationField

In [0]:
%sql
select count(employeecount) from emp_attrition;

count(employeecount) 1470

In [0]:
%sql
select Attrition, count(Attrition)as AttrCount from emp_attrition
group by Attrition

Attrition AttrCount No 1233 Yes 237

Databricks visualization. Run in Databricks to view.

In [0]:

%sql
With AttritionCount (select Attrition, count(Attrition)as AttrCount from emp_attrition
group by Attrition)
select * from AttritionCount where Attrition = "Yes";

Attrition AttrCount Yes 237

In [0]:
%sql

with final as (
  select age, sum(employeecount) as sumofempcount,
  case 
    when age between 18 and 24 then '18-24'
    when age between 25 and 31 then '25-31'
    when age between 32 and 38 then '32-38'
    when age between 39 and 45 then '39-45'
    when age between 46 and 52 then '46-52'
    else '52+'
  end as age_group
  from emp_attrition
  where Attrition = 'Yes'
  group by age, age_group
)
select age_group, sum(sumofempcount)
from final
group by age_group

age_group sum(sumofempcount) 39-45 29 46-52 21 32-38 56 18-24 38 25-31 80 52+ 13

Databricks visualization. Run in Databricks to view.

In [0]:
%sql

-- select * from emp_attrition limit 10

select department, count(department) from emp_attrition
where Attrition == 'Yes'
group by department;

department count(department) Sales 92 Research & Development 133 Human Resources 12

Databricks visualization. Run in Databricks to view.

In [0]:
%sql
select * from emp_attrition limit 1;
select distinct(BusinessTravel) from emp_attrition;

select BusinessTravel, sum(employeecount) from emp_attrition
where Attrition == 'Yes'
group by BusinessTravel;

BusinessTravel sum(employeecount) Travel_Frequently 69 Non-Travel 12 Travel_Rarely 156

Databricks visualization. Run in Databricks to view.

In [0]:
%sql
select * from emp_attrition limit 1;

-- check distribution of daily rate
select distinct(DailyRate) from emp_attrition;
select max(distinct(DailyRate)) from emp_attrition;

-- spliting daily rate (1-500 as LOW | 501-1000 as MID | 1001-1500 as HIGH)

with cte (select DailyRate, sum(employeecount) as sumofempcount,
  case 
    when DailyRate between 0 and 300 then 'LOW'
    when DailyRate between 501 and 1000 then 'MID'
    else 'HIGH'
  end as DailyRateGroup
  from emp_attrition
  where Attrition = 'Yes'
  group by DailyRateGroup, DailyRate)
select DailyRateGroup, sum(sumofempcount)
from cte
group by DailyRateGroup;


DailyRateGroup sum(sumofempcount) MID 88 HIGH 115 LOW 34

Databricks visualization. Run in Databricks to view.

In [0]:
%sql
select * from emp_attrition limit 1;
select distinct(gender) from emp_attrition;

with Gender (select gender, count(gender) as total_gender from emp_attrition
where Attrition == 'Yes'
group by gender
order by total_gender)
select gender, total_gender, round((total_gender / 237) * 100, 1) as Percentage
from Gender;


gender total_gender Percentage Female 87 36.7 Male 150 63.3

Databricks visualization. Run in Databricks to view.

In [0]:

%sql
-- Education
--    1 - No College
--    2 - College
--    3 - Bachelor Degree
--    4 - Master Degree
--    5 - PhD

With cte (select education, count(employeenumber) as total_employees from emp_attrition
where Attrition == 'Yes'
group by education
order by total_employees)
select
  case
    when education == 1 then "No College Degree"
    when education == 2 then "College Degree"
    when education == 3 then "Bachelor Degree"
    when education == 4 then "Master Degree"
    else "PhD"
  end as EducationLevel,
total_employees from cte;


EducationLevel total_employees PhD 5 No College Degree 31 College Degree 44 Master Degree 58 Bachelor Degree 99

Databricks visualization. Run in Databricks to view.

In [0]:

%sql
-- (1-low, 2-medium, 3-high, 4-highly satisfied)
select EnvironmentSatisfaction, count(employeenumber) as total_employees from emp_attrition
where attrition='Yes'
group by 1
order by 1 desc;

EnvironmentSatisfaction total_employees 4 60 3 62 2 43 1 72

Databricks visualization. Run in Databricks to view.

In [0]:

%sql
-- (1-low, 2-medium, 3-high, 4-highly satisfied)
select distinct(jobsatisfaction) from emp_attrition;

with cte (select jobsatisfaction, count(employeenumber) as total_employees from emp_attrition
where attrition='Yes'
group by 1
order by 1 desc)
select
  case
    when jobsatisfaction == 1 then "low"
    when jobsatisfaction == 2 then "medium"
    when jobsatisfaction == 3 then "high"
    else "highly satisfied"
  end as EducationLevel,
total_employees from cte;


EducationLevel total_employees highly satisfied 52 high 73 medium 46 low 66

Databricks visualization. Run in Databricks to view.

In [0]:
%sql
select MaritalStatus, count(employeenumber) as total_employees from emp_attrition
where attrition='Yes'
group by 1
order by 1 desc;

MaritalStatus total_employees Single 120 Married 84 Divorced 33

Databricks visualization. Run in Databricks to view.

In [0]:
%sql
select MaritalStatus, Gender, count(employeenumber) as total_employees from emp_attrition
where attrition='Yes'
group by 1, 2
order by 1;

MaritalStatus Gender total_employees Divorced Female 9 Divorced Male 24 Married Female 31 Married Male 53 Single Male 73 Single Female 47

Databricks visualization. Run in Databricks to view.

In [0]:
%sql
select * from emp_attrition limit 5;

Age Attrition BusinessTravel DailyRate Department DistanceFromHome Education EducationField EmployeeCount EmployeeNumber EnvironmentSatisfaction Gender HourlyRate JobInvolvement JobLevel JobRole JobSatisfaction MaritalStatus MonthlyIncome MonthlyRate NumCompaniesWorked Over18 OverTime PercentSalaryHike PerformanceRating RelationshipSatisfaction StandardHours StockOptionLevel TotalWorkingYears TrainingTimesLastYear WorkLifeBalance YearsAtCompany YearsInCurrentRole YearsSinceLastPromotion YearsWithCurrManager 41 Yes Travel_Rarely 1102 Sales 1 2 Life Sciences 1 1 2 Female 94 3 2 Sales Executive 4 Single 5993 19479 8 Y Yes 11 3 1 80 0 8 0 1 6 4 0 5 49 No Travel_Frequently 279 Research & Development 8 1 Life Sciences 1 2 3 Male 61 2 2 Research Scientist 2 Married 5130 24907 1 Y No 23 4 4 80 1 10 3 3 10 7 1 7 37 Yes Travel_Rarely 1373 Research & Development 2 2 Other 1 4 4 Male 92 2 1 Laboratory Technician 3 Single 2090 2396 6 Y Yes 15 3 2 80 0 7 3 3 0 0 0 0 33 No Travel_Frequently 1392 Research & Development 3 4 Life Sciences 1 5 4 Female 56 3 1 Research Scientist 3 Married 2909 23159 1 Y Yes 11 3 3 80 0 8 3 3 8 7 3 0 27 No Travel_Rarely 591 Research & Development 2 1 Medical 1 7 1 Male 40 3 1 Laboratory Technician 2 Married 3468 16632 9 Y No 12 3 4 80 1 6 3 3 2 2 2 2

Insights drawn

1. Employee Aged 26 to 32 are leaving organization
2. Employees from the Research and Develoment department are leaving the organization
3. 41.8% employees with bachelor degree left organization
4. 72 employes that are not satisfied with the environment left the organization
5. 156 employees that rarely travels left organization